bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ErrorUtils.java

```diff
@@ -143,4 +145,16 @@ public static BError createUnorderedTypesError(Object lhsValue, Object rhsValue)
         throw createError(BallerinaErrorReasons.UNORDERED_TYPES_ERROR, BLangExceptionHelper.getErrorMessage(
                 RuntimeErrors.UNORDERED_TYPES_IN_COMPARISON, lhsValue, rhsValue));
     }
+
+    public static BError createConversionError(Object inputValue, Type targetType) {
+        return createError(VALUE_LANG_LIB_CONVERSION_ERROR,
+                BLangExceptionHelper.getErrorMessage(INCOMPATIBLE_CONVERT_OPERATION,
+                        TypeChecker.getType(inputValue), targetType));
+    }
+
+    public static BError createConversionError(Object inputValue, Type targetType, String detailMessage) {
+        return createError(VALUE_LANG_LIB_CONVERSION_ERROR, BLangExceptionHelper.getErrorMessage(
+                INCOMPATIBLE_CONVERT_OPERATION, TypeChecker.getType(inputValue), targetType)
+                .concat(StringUtils.fromString(": ".concat(detailMessage))));
```

Comment:
Shall we use `+` instead of the `concat` calls? They are more readable.

Reply:
I will look into all these.

Reply:
I used + with java Strings, but + can cannot be used with BStrings and also cannot pass detailMessage into the BLangExceptionHelper.getErrorMessage() to get the required output. So kept that as concat.